### ISSUE TITLE
Suggestion: "Tutorial"

### ISSUE THREAD

**Issue by dakom** [CosmWasm/docs#155]
The current docs structure is excellent for someone who is working with CosmWasm and needs to look up a specific topic (e.g. how to work with wasmd, storey, add ibc hooks, etc.), or someone brand-new to CosmWasm and wants a tour of everything it can do.

However, we received a couple comments - and I agree, that it's not very clear from a "how to get started, hands-on" perspective.

What about a separate section called "Tutorial" that is a start-to-finish guide with to create a contract, deploy it, test it, etc.?

Where applicable, it can show examples with different tools in tabs (like the "Containers" section in Storey showing storage-plus vs. Story: https://docs.cosmwasm.com/storey/containers)

**Comment by jawoznia** [comment]
I agree. I created the #111 issue for that as I also am missing the actual usage of CosmWasm. It's good that the documentation is atomized to the semantical parts, but we are missing the bigger picture of how to use our product.

**Comment by jawoznia** [comment]
I have been thinking about it and we could make a section `Examples` in the docs. There we could through simple contracts like `Counter`, two contracts communicating, usage of `CustomQuery` and `CustomMsg`, and so on.

We could then for one of them present the full path from creation through deployment to communication with the contract stored on the chain.

**Comment by hashedone** [comment]
What do you think about starting with moving the book (https://book.cosmwasm.com/, https://cosmwasm.github.io/sylvia-book/index.html) to the docs portal? Maybe not copy-paste, but refresh it in the process. It was written with such an approach in mind.

**Comment by kulikthebird** [comment]
Yep, it would be good to have single source of truth. Easier to maintain for us, better for users to look for info
